x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrRepositoryManager.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ccr;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.admin.cluster.repositories.delete.DeleteInternalRepositoryAction;
+import org.elasticsearch.action.admin.cluster.repositories.delete.DeleteInternalRepositoryRequest;
+import org.elasticsearch.action.admin.cluster.repositories.put.PutInternalRepositoryAction;
+import org.elasticsearch.action.admin.cluster.repositories.put.PutInternalRepositoryRequest;
+import org.elasticsearch.action.support.PlainActionFuture;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.ConcurrentCollections;
+import org.elasticsearch.transport.RemoteClusterAware;
+import org.elasticsearch.xpack.ccr.repository.CcrRepository;
+
+import java.util.List;
+import java.util.Set;
+
+class CcrRepositoryManager extends RemoteClusterAware {
+
+    private final NodeClient client;
+    private final Set<String> clusters = ConcurrentCollections.newConcurrentSet();
```

Comment:
Since `put` and `delete` should be idempotent (and they other for other normal repositories requests) I removed the cache and just call the action each time. 

The other option we to create a `get` action, but I thought that was unnecessary as we need to handle potential concurrency in the `put` and `delete` methods anyway.

Reply:
>  I removed the cache and just call the action each time.

sounds good